core/src/main/scala/chisel3/properties/Property.scala

```diff
@@ -348,6 +349,55 @@ object Property {
     // This cast should be safe, because there are no members of prop.ClassType to access
     def as(prop: Property[ClassType]): Property[ClassType] with prop.ClassType =
       prop.asInstanceOf[Property[ClassType] with prop.ClassType]
+
+    /** Connect a reference to an Instance[Class] to a Property[ClassType] prop.
+      *
+      * This method allows Instances of Classes to be safely connected to Property[ClassType] ports, so the references
+      * can be passed through the hierarchy.
+      *
+      * @param source An Instance[T <: Class], where T is the same class as prop's ClassType.
+      * @param sourceInfo The source location where the connection is made, for error messages.
+      */
+    def :=[T <: Class](source: Instance[T])(implicit sourceInfo: SourceInfo) = {
```

Comment:
When you put it like that, it does feel awkward. I guess what we want here is an extension method on `Instance` like you're suggesting, which converts `Instance[Class]` to the `Property[ClassType]` for that class, and then the normal `:=` for Properties works.

Reply:
> When you put it like that, it does feel awkward. I guess what we want here is an extension method on Instance like you're suggesting, which converts Instance[Class] to the Property[ClassType] for that class, and then the normal := for Properties works.

Yes, that what I mean, but since we don't do the release for `Property` recently, I think we can safely get this in, and do the API change in the following PRs..

Reply:
Yeah, but since this is a new API I'd also prefer to add it in the shape we really want moving forward. I'll take a look and if it's not too hard update it in this PR.

Reply:
Sure! let's move fast! :)